src/mainboard/clevo/mtl-h/variants/dgpu/ramstage.c

```diff
@@ -2,10 +2,41 @@
 
 #include <mainboard/variants.h>
 #include <gpio.h>
+#include <dasharo/options.h>
+#include <device/pci_ids.h>
+#include <device/pci_ops.h>
+#include <device/pci_def.h>
+#include <soc/intel/common/reset.h>
 
 void variant_devtree_update(void)
 {
 	struct device *i2c_amp_dev = pcidev_on_root(0x19, 1);
 	bool have_smartamp = !gpio_get(GPP_E00);
 	i2c_amp_dev->enabled = have_smartamp;
+
+	struct device *dgpu_dev = pcidev_on_root(0x01, 0);
+	dgpu_dev->enabled = dasharo_is_dgpu_enabled() != 0;
+	printk(BIOS_DEBUG, "dgpu_dev->enabled: %d\n", dgpu_dev->enabled);
```

Comment:
possible null pointer deref

```suggestion
	struct device *dgpu_dev = pcidev_on_root(0x01, 0);
	if (dgpu_dev) {
	        dgpu_dev->enabled = dasharo_is_dgpu_enabled() != 0;
	        printk(BIOS_DEBUG, "dgpu_dev->enabled: %d\n", dgpu_dev->enabled);    
	}
```

Reply:
Same applies to the i2c amp dev above actually, can you fix that one as well?

Reply:
[Done](https://github.com/Dasharo/coreboot/pull/630/commits/fe4cc24a49b692d4358aa8d196513a478d93f995)